src/Microsoft.AspNet.OData.Shared/Routing/Conventions/AttributeRoutingConvention.cs

```diff
@@ -59,6 +59,11 @@ private AttributeRoutingConvention(string routeName)
 
                     return result;
                 }
+
+                // It's possible that template.TryMatch inserted values in the values dict even if
```

Comment:
It's certain ... , right?

Reply:
It's no certain. It depends on whether `TryMatch` was called on path segments that have values, for example `KeySegment`. But if  `TryMatch` was only called on `EntitySetSegment`, then it would not add values to the dict.